components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,193 @@
+import pg from "pg";
+
 export default {
   type: "app",
   app: "postgresql",
-  propDefinitions: {},
+  propDefinitions: {
+    table: {
+      type: "string",
+      label: "Table",
+      description: "Database table",
+      async options() {
+        return this.getTables();
+      },
+    },
+    column: {
+      type: "string",
+      label: "Column",
+      description: "The name of a column in the table to use for deduplication. Defaults to the table's primary key",
+      async options({ table }) {
+        return this.getColumns(table);
+      },
+    },
+    query: {
+      type: "string",
+      label: "SQL Query",
+      description: "Your custom SQL query",
+    },
+    value: {
+      type: "string",
+      label: "Lookup Value",
+      description: "Value to search for",
+      async options({
+        table, column,
+      }) {
+        return this.getColumnValues(table, column);
+      },
+    },
+    rowValues: {
+      type: "object",
+      label: "Row Values",
+      description: "Enter the column names and respective values as key/value pairs",
```

Comment:
I guess here we can also give a clue to the user on how to add the key/value pairs when structure mode is off describing the value as `{{{columnName:"columnValue"}}}`?